dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -128,39 +129,79 @@ public WorkflowResource(HttpClient client, UserDAO userDAO, TokenDAO tokenDAO, W
     public void refreshStubWorkflowsForUser(User user) {
         try {
             List<Token> tokens = checkOnBitbucketToken(user);
-            // TODO: do this for bitbucket as well as for github
-            // Token bitbucketToken = Helper.extractToken(tokens, TokenType.BITBUCKET_ORG.toString());
-            Token githubToken = Helper.extractToken(tokens, TokenType.GITHUB_COM.toString());
 
-            // get workflows from github for a user, experiment with github first
-            //final SourceCodeRepoInterface sourceCodeRepo = SourceCodeRepoFactory.createSourceCodeRepo(null, client,
-            //    bitbucketToken == null ? null : bitbucketToken.getContent(), githubToken.getContent());
-            if (githubToken == null || githubToken.getContent() == null){
-                return;
-            }
-            final SourceCodeRepoInterface sourceCodeRepo = new GitHubSourceCodeRepo(user.getUsername(),githubToken.getContent(), null);
-            final Map<String, String> workflowGitUrl2Name = sourceCodeRepo.getWorkflowGitUrl2RepositoryId();
-            for(Map.Entry<String, String> entry : workflowGitUrl2Name.entrySet()) {
-                final List<Workflow> byGitUrl = workflowDAO.findByGitUrl(entry.getKey());
-                if (byGitUrl.size() > 0) {
-                    for (Workflow workflow : byGitUrl) {
-                        // when 1) workflows are already known, update the copy in the db
-                        // update the one workflow from github
-                        final Workflow newWorkflow = sourceCodeRepo.getNewWorkflow(entry.getValue(), Optional.of(workflow));
-                        // take ownership of these workflows
-                        workflow.getUsers().add(user);
-                        updateDBWorkflowWithSourceControlWorkflow(workflow, newWorkflow);
+            // Refresh Bitbucket
```

Comment:
This section looks like a copy-paste of the next added section with the different being that it uses bitbucketSourceCodeRepo to getNewWorkflow. Try to remove this duplication by determining which interface to use based on the git url or at least extract a method that takes SourceCodeRepoInterface as a parameter.